opm/autodiff/ThreadHandle.hpp

```diff
@@ -53,13 +53,28 @@ namespace Opm
       // no copying
       ThreadHandleQueue( const ThreadHandleQueue& ) = delete;
 
+      // wait duration of 10 milli seconds
+      void wait() const
+      {
+          std::this_thread::sleep_for( std::chrono::milliseconds(10) );
+      }
+
     public:
       //! constructor creating object that is executed by thread
       ThreadHandleQueue()
         : objQueue_(), mutex_()
       {
       }
 
+      ~ThreadHandleQueue()
+      {
+        // wait until all objects have been written.
+        while( ! objQueue_.empty() )
+        {
+            wait();
+        }
+      }
+
```

Comment:
This should prevent the program from terminating, unless all data was written. Please also see the method run which is actually an infinite loop.